src/nats.h

```diff
@@ -4564,6 +4564,17 @@ natsSubscription_AutoUnsubscribe(natsSubscription *sub, int max);
 NATS_EXTERN natsStatus
 natsSubscription_QueuedMsgs(natsSubscription *sub, uint64_t *queuedMsgs);
 
+/** \brief Returns the subscription id and subject name.
+ *
+ * Returns the subscription id and subject name of this subscription.
+ *
+ * @param sub the pointer to the #natsSubscription object.
+ * @param sid if not `NULL`, the memory location where to store the subscription id.
+ * @param subject if not `NULL`, the memory location where to store the subject name.
+ */
+NATS_EXTERN natsStatus
+natsSubscription_GetInfo(natsSubscription* sub, int64_t* sid, char* subject);
```

Comment:
Thank you for adding the doc! Since it looks like you are returning the subject from the subscription object, and not a copy, I usually make that a `const char*` and add a `\warning` in the doc to indicate that the returned string belongs to the subscription object and the user must not free this memory.
Here is an example for natsMsg_GetSubject: https://nats-io.github.io/nats.c/group__msg_group.html#gad7cad38946648e3047aa0665a3dd230a

Reply:
Valid remark. Changed to const and added the warning in the docs.